app/assets/javascripts/admin/addon/components/admin-config-areas/components.gjs

```diff
@@ -45,19 +93,428 @@ export default class AdminConfigAreasComponents extends Component {
       },
       duration: 2000,
     });
-    this.router.refresh();
+    this.load();
+  }
+
+  @action
+  onNameFilterChange(event) {
+    this.loading = true;
+    this.nameFilter = event.target.value;
+    discourseDebounce(this, this.load, INPUT_DELAY);
+  }
+
+  @action
+  onStatusFilterChange(value) {
+    this.statusFilter = value;
+    this.load();
+  }
+
+  @action
+  async load() {
+    this.loading = true;
+
+    try {
+      const data = await ajax("/admin/config/customize/components", {
+        data: { name: this.nameFilter, status: this.statusFilter },
+      });
+
+      this.components = data.components;
+    } finally {
+      this.loading = false;
+    }
   }
 
   <template>
-    <div class="admin-detail">
-      <ThemesGrid @themes={{@components}}>
-        <:specialCard>
-          <InstallThemeCard
-            @component={{true}}
-            @openModal={{this.installModal}}
+    <DPageSubheader
+      @titleLabel={{i18n
+        "admin.config_areas.themes_and_components.components.title"
+      }}
+      @descriptionLabel={{i18n
+        "admin.config_areas.themes_and_components.components.description"
+      }}
+    >
+      <:actions as |actions|>
+        <actions.Primary
+          disabled={{this.loading}}
+          @label="admin.config_areas.themes_and_components.components.install"
+          @action={{this.installModal}}
+        />
+      </:actions>
+    </DPageSubheader>
+    <div class="container">
+      <div class="admin-config-components__filters">
```

Comment:
<img width="1117" alt="Screenshot 2025-04-04 at 4 08 02 PM" src="https://github.com/user-attachments/assets/b389df5f-aefa-417b-9dde-54158bed8688" />

When there are not components installed, I don't think we should render these filters.